File: src/main/org/nlogo/prim/_list.java
package org.nlogo.prim ;

import org.nlogo.api.LogoException;
import org.nlogo.api.LogoListBuilder;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _list
	extends Reporter
	implements org.nlogo.nvm.Pure, org.nlogo.nvm.CustomGenerated
{
	@Override
	public Syntax syntax()
	{
		int[] right = { Syntax.TYPE_REPEATABLE | Syntax.TYPE_WILDCARD } ;
		int ret = Syntax.TYPE_LIST ;
		return Syntax.reporterSyntax( right , ret , 2 , 0 ) ;
	}
	@Override
	public Object report ( final Context context ) throws LogoException
	{
		LogoListBuilder list = new LogoListBuilder() ;
		for( int i = 0 ; i < args.length ; i ++ ) 
		{
			list.add( args[ i ].report( context ) ) ;
		}
		return list.toLogoList() ;
	}
}

File: headless/src/main/org/nlogo/prim/_list.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.{ LogoException, LogoListBuilder, Syntax }
import org.nlogo.nvm.{ Reporter, Context, Pure, CustomGenerated }

class _list extends Reporter with Pure with CustomGenerated {
  override def syntax =
    Syntax.reporterSyntax(
      Array(Syntax.RepeatableType | Syntax.WildcardType),
      Syntax.ListType, 2, 0)
  override def report(context: Context) = {
    val builder = new LogoListBuilder
    var i = 0
    while(i < args.length) {
      builder.add(args(i).report(context))
      i += 1
    }
    builder.toLogoList
  }
}
